index.js

```diff
@@ -723,13 +723,17 @@ function asciiSlice (buf, start, end) {
 }
 
 function binarySlice (buf, start, end) {
-  var ret = ''
-  end = Math.min(buf.length, end)
```

Comment:
Will re-add this,  shouldn't have been removed. 
Probably need to add a test for it.